### PR TITLE
Feature specific extends

### PR DESCRIPTION
**Pull request by JustBeard**
Please check README. All new changes described there.
Also you can see WebApp migration here: https://github.com/EkoCommunications/EkoWebApp2/pull/301

UPD:
- https://github.com/EkoCommunications/EkoESLintConfig/pull/1#discussion_r250263987
- removed import/no-unresolved as 'ignore' hack is not required anymore;
- update comment for import/prefer-default-export rule;
- describe how to avoid issue with using babel root import synonym (in docs, Hint section)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,134 +1,5 @@
...
-    'no-prototype-builtins': 'off',
-
-    // Ref: https://eslint.org/docs/rules/consistent-return
-    // Airbnb: 'consistent-return': 'error'
-    //
-    // TODO: Until we decide, do we want to do `return null` explicitly from function or we stick with
-    // `return undefined`, we disable this rule.
-    'consistent-return': 'off',
-    // ========== END CORE ==========
+const ekoRules = [require.resolve('./rules/base'), require.resolve('./rules/import')];
```
Comment (tavurth) [inline index.js]: Here's a great example of how we can use prettier's comment approach to expand this array

```suggestion
const ekoRules = [
  // Comment about what each of these imports are for
  require.resolve('./rules/base'), 
  require.resolve('./rules/import')
];
```

File: package.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "eslint-config-eko",
-  "version": "0.2.3",
+  "version": "1.0.0",
```
Comment (tavurth) [inline package.json:3]: Suggest version 0.9

**Review by tavurth — CHANGES_REQUESTED** [review]

**Comment by tavurth** [comment]
Looks good 👍 Nicely split code

File: rules/import.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    // Enforces the addition of file extensions when importing.
+    'import/extensions': [
+      'error',
+      'never',
+      {
+        mp3: 'always',
+        mp4: 'always',
+        svg: 'always',
+        png: 'always',
+      },
```
Comment (beermindnor) [inline rules/import.js]: how about (.jpg, .jpeg, css, scss, sass, less)

File: package.json (hunk 2 above)
Comment (JustBeard) [reply package.json:3]: @tavurth  I though to use `1.0.0` as there are breaking changes. And update will require manual update configurations

**Review by spiritpun — APPROVED** [review]

File: rules/import.js (hunk 3 above)
Comment (JustBeard) [reply rules/import.js]: Actually I did not think it thru. Just made it on use cases what we have.
fiy, this is airbnb
```
    'import/extensions': ['error', 'ignorePackages', {
      js: 'never',
      mjs: 'never',
      jsx: 'never',
    }],
```
Might we should go with this. @beermindnor @tavurth 
```
    'import/extensions': ['error', 'always', {
      js: 'never',
      mjs: 'never',
      jsx: 'never',
    }],
```

Comment (beermindnor) [reply rules/import.js]: the second, does it mean we need to put extension for dependency?
import BarLoader from 'loaders/barloader.js'; ? if yes then we should go with airbnb?

**Review by tavurth — APPROVED** [review]

**Review by Jedmed — APPROVED** [review]

File: rules/import.js (hunk 3 above)
Comment (JustBeard) [reply rules/import.js]: Second option says that we always define extension on import except 'js', 'jsx', 'mjs'.

Comment (JustBeard) [reply rules/import.js]: Jeez, this rule is mess. "always" option is actually working a little bit weird.
https://github.com/benmosher/eslint-plugin-import/issues/414 - here is an explanation.
So it seems that we should use default airbnb configuration. I am gonna remove this rule from our set

File: README.md [hunk 4]
```diff
@@ -34,3 +66,57 @@ In package.json:
     }
   },
 ```
+
+## Hints
+
+### import/no-unresolved and root path synonym
+
+If you want to use root synonyms like:
+
```
Comment (spiritpun) [inline README.md:75]: Nice one.

**Review by spiritpun — APPROVED** [review]
LGTM

**Review by beermindnor — APPROVED** [review]
Many thanks for those painful 🔥changes .

**Pull request merged**